opal/mca/threads/pthreads/threads_pthreads_tsd.h

```diff
@@ -32,6 +32,24 @@
 
 typedef pthread_key_t opal_tsd_key_t;
 
+typedef struct _opal_tsd_tracked_key_t opal_tsd_tracked_key_t;
+
+typedef struct _opal_tsd_data_list_t {
+    opal_list_item_t super;
+    opal_tsd_tracked_key_t *tracked_key;
+    void *data;
+} opal_tsd_data_list_t;
+OBJ_CLASS_DECLARATION(opal_tsd_data_list_t);
+
+struct _opal_tsd_tracked_key_t {
+    opal_object_t super;
+    opal_tsd_key_t key;
+    opal_mutex_t mutex;
+    opal_list_t tsd_list;
+    void (*user_destructor)(void *);
+};
+OBJ_CLASS_DECLARATION(opal_tsd_tracked_key_t);
+
```

Comment:
Why is this in the pthread_tsd and not in tsd.h?

This API should fully rely on opal_tsd_key_t and thus not be thread implementation-specific.